test/fhirdefs/load.test.ts

```diff
@@ -1,427 +1,28 @@
-              version: '4.0.0',
-              tool: '4.1.0 (3)',
-              repo: 'HL7/US-Core-R4/branches/newbranch/qa.json'
-            },
-            {
-              url: 'http://hl7.org/fhir/sushi-test-no-download/ImplementationGuide/sushi-test-no-download-0.1.0',
-              name: 'sushi-test-no-download',
-              'package-id': 'sushi-test-no-download',
-              'ig-ver': '0.1.0',
-              repo: 'sushi/sushi-test-no-download/branches/master/qa.json'
-            },
-            {
-              url: 'http://hl7.org/fhir/sushi-test-old/ImplementationGuide/sushi-test-old-0.1.0',
-              name: 'sushi-test-old',
-              'package-id': 'sushi-test-old',
-              'ig-ver': '0.1.0',
-              repo: 'sushi/sushi-test-old/branches/master/qa.json'
-            },
-            {
-              url: 'http://hl7.org/fhir/sushi-test/ImplementationGuide/sushi-test-0.1.0',
-              name: 'sushi-test',
-              'package-id': 'sushi-test',
-              'ig-ver': '0.1.0',
-              repo: 'sushi/sushi-test/branches/master/qa.json'
-            }
-          ]
-        };
-      } else if (
-        uri === 'https://build.fhir.org/ig/HL7/US-Core-R4/package.manifest.json' ||
-        (uri.startsWith('https://build.fhir.org/ig/sushi/sushi-test') && uri.endsWith('json'))
-      ) {
-        return {
-          data: {
-            date: '20200413230227'
-          }
-        };
-      } else if (
-        uri === 'https://packages.fhir.org/sushi-test/0.2.0' ||
-        uri === 'https://build.fhir.org/ig/sushi/sushi-test-old/package.tgz' ||
-        uri === 'https://build.fhir.org/ig/HL7/US-Core-R4/package.tgz' ||
-        uri === 'https://build.fhir.org/hl7.fhir.r5.core.tgz' ||
-        uri === 'https://packages2.fhir.org/packages/hl7.fhir.r4b.core/4.1.0' ||
-        uri === 'https://packages.fhir.org/hl7.fhir.r4b.core/4.3.0' ||
-        uri === 'https://packages2.fhir.org/packages/hl7.fhir.r5.core/4.5.0' ||
-        uri === 'https://packages.fhir.org/hl7.fhir.r4.core/4.0.1' ||
-        uri === 'https://packages2.fhir.org/packages/fhir.dicom/2021.4.20210910'
-      ) {
-        return {
-          data: {
-            some: 'zipfile'
-          }
-        };
-      } else if (
-        uri === 'https://packages.fhir.org/hl7.fhir.r4b.core/4.1.0' ||
-        uri === 'https://packages.fhir.org/hl7.fhir.r5.core/4.5.0' ||
-        uri === 'https://packages.fhir.org/fhir.dicom/2021.4.20210910'
-      ) {
-        throw 'Not Found';
-      } else {
-        return {};
+jest.mock('fhir-package-loader', () => {
+  const original = jest.requireActual('fhir-package-loader');
+  return {
+    ...original,
+    loadDependency: jest.fn(
+      async (packageName: string, version: string, FHIRDefs: FHIRDefinitions) => {
+        // the mock loader can find R2, R3, and R5
+        if (/hl7\.fhir\.r(2|3|5).core/.test(packageName)) {
+          loadedPackages.push(`${packageName}#${version}`);
```

Comment:
As far as I can tell, these tests don't ever actually read the `loadedPackages` variable?  Should they?  If yes, then I guess the tests need to be updated to do that.  If no, then I think `loadedPackages` can be removed from this file entirely.

Reply:
Oops I didn't realize this was never used (and that it was using `let` above). But you're right, I took out the check where this would have been used (and didn't realize that was the only place that this was used in this file). The problem was that in the old places it would have been used ([here](https://github.com/FHIR/sushi/pull/1026/commits/6f38ef12511fedb585b2aa6d2125d9404955a6ea#diff-72b4539a412a1bf5daa36f8e26f20b222a0cdc6ef59eecb42038162acc2f2bf5L174), [here](https://github.com/FHIR/sushi/pull/1026/commits/6f38ef12511fedb585b2aa6d2125d9404955a6ea#diff-72b4539a412a1bf5daa36f8e26f20b222a0cdc6ef59eecb42038162acc2f2bf5L191), and [here](https://github.com/FHIR/sushi/pull/1026/commits/6f38ef12511fedb585b2aa6d2125d9404955a6ea#diff-72b4539a412a1bf5daa36f8e26f20b222a0cdc6ef59eecb42038162acc2f2bf5L206)), it was testing that the package was actually added to `supplementalFHIRPackages` and not added to the old `packages` list. But now that the `packages` list doesn't exist on `FHIRDefinitions` and the package is added to the static `loadedPackages` list any time a package is "loaded", there didn't seem to be anything to test for the `pacakges`/`loadedPackages` side of things. Instead, the tests just check that the package is added to `supplementalFHIRPackages`.

I can remove the `loadedPackages` property entirely. But if you disagree or think there's something I can test that I'm missing, let me know.

Reply:
Removed in [377ba4f](https://github.com/FHIR/sushi/pull/1026/commits/377ba4f0f70bbf9b2226773fb692d3662cde7211) but I can update if you have other idea from the above novel I wrote.